src/ziggurat/streams.clj

```diff
@@ -54,13 +61,13 @@
 (defn- map-values [mapper-fn stream-builder]
   (.mapValues stream-builder (value-mapper mapper-fn)))
 
-(defn- transformer-supplier [metric-namespace]
+(defn- transformer-supplier [metric-namespace oldest-processed-message-in-s]
   (reify TransformerSupplier
-    (get [_] (kafka-delay/create-transformer metric-namespace))))
+    (get [_] (transformer/create metric-namespace oldest-processed-message-in-s))))
 
-(defn- transform-values [topic-entity stream-builder]
+(defn- transform-values [topic-entity skip-before-time stream-builder]
   (let [metric-namespace (get-metric-namespace "message-received-delay-histogram" topic-entity)]
-    (.transform stream-builder (transformer-supplier metric-namespace) (into-array [(.name (store-supplier-builder))]))))
+    (.transform stream-builder (transformer-supplier metric-namespace skip-before-time) (into-array [(.name (store-supplier-builder))]))))
```

Comment:
why does the variable `skip-before-time` have different names in different functions. It gets confusing and complicated

Reply:
Changed it to make it uniform